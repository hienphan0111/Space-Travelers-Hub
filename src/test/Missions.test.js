import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect.js';
import { Provider } from 'react-redux';
import store from 'redux/store';
import Missions from 'routes/missions.route';
import axios from 'axios';

jest.mock('axios', () => ({
  post: jest.fn(),
  get: jest.fn()
}))

describe('Render the missions lists', () => {
  it('Renders the missions correctly', () => {
    const mission = {
      id: 2,
      mission_name: 'one',
      description: 'a mission',
      joined: true,
    };
    const missions = TestRenderer.create(
      <Provider store={store}>
        <Missions key={mission.id} missions={mission}/>
      </Provider>,
    ).toJSON();
    expect(missions).toMatchSnapshot();
  });
});