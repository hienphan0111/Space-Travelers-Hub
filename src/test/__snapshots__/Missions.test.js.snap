// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the missions lists Renders the missions correctly 1`] = `
<div
  className="missions-container"
>
  <div
    className="header table-template"
  >
    <span>
      Mission
    </span>
    <span>
      Description
    </span>
    <span>
      Status
    </span>
    <span />
  </div>
</div>
`;
